Packs/Kafka/Integrations/KafkaV3/KafkaV3.py

```diff
@@ -826,13 +829,21 @@ def fetch_incidents(kafka: KafkaCommunicator, demisto_params: dict) -> None:
             kafka_consumer.assign(topic_partitions)
 
             demisto.debug("Beginning to poll messages from kafka")
-
+            num_polled_msg = 0
             for _ in range(max_messages):
-                polled_msg = kafka_consumer.poll(kafka.POLL_TIMEOUT)
+                # Initial message consumption may take up to
+                # `session.timeout.ms` for the consumer group to
+                # rebalance and start consuming
+                polled_msg = kafka_consumer.poll(poll_timeout)
                 if polled_msg:
-                    demisto.debug("Received a message from Kafka.")
+                    num_polled_msg += 1
+                    demisto.debug(f"Received a message {num_polled_msg}# from Kafka.")
                     incidents.append(create_incident(message=polled_msg, topic=topic))
                     last_fetched_offsets[f'{polled_msg.partition()}'] = polled_msg.offset()
+                elif stop_consuming_upon_timeout and (not polled_msg):
```

Comment:
If we are just validating this is not the first message which we are getting timeout on, we can change the num+polled_msg to be a boolean var.